src/elements/types.rs

```diff
@@ -177,8 +177,9 @@ impl Deserialize for FunctionType {
 
 		let params: Vec<ValueType> = CountedList::deserialize(reader)?.into_inner();
 
-		let has_return_type = VarUint1::deserialize(reader)?;
-		let return_type = if has_return_type.into() {
+		let return_types: u32 = VarUint32::deserialize(reader)?.into();
+
+		let return_type = if return_types > 0 {
```

Comment:
Let's return an error if `return_types` isn't equal to 0 nor 1 ?

Reply:
indeed